src/libraries/System.Private.CoreLib/src/System/DefaultBinder.cs

```diff
@@ -383,8 +383,10 @@ internal partial class DefaultBinder : Binder
 #endregion
             }
 
+            MethodBase bestMatch = candidates[currentMin]!;
+
             if (ambig)
-                throw new AmbiguousMatchException();
+                throw ThrowHelper.GetAmbiguousMatchException(bestMatch);
```

Comment:
NIT: the name `bestMatch` doesn't make much sense for throwing cases, could use something like `match` or if the name already used then something like `foundMatch`

Reply:
Hm, I see it was already using this name in other places, never mind